### PR TITLE
Integration Tests (first pass)

### PR DESCRIPTION
**Pull request by jasonLaster**
This adds basic support for end-to-end integration tests.

How does it work:

1. The debugger client is opened in an iframe inside of a mocha web runner.
The tests are actually very similar to our unit tests that run in a web
context. The debugger is driven by a tool called cypress that performs
all of the actions like clicking, typing, etc...

2. The debuggee is open in firefox with selenium web driver. Commands
are sent to the debuggee over HTTP with a simple express server.

Next Steps:

1. Add a headless runner (cypress-cli) should not be too hard.
2. Add to circle ci. Also should not be too hard
3. Write tests for debugger commands (resume, step in, etc). The fun
part

Trying it out:

1. Run selenium server beforehand java -jar ~/bin/selenium-server-standalone-2.53.0.jar
2. Run `node bin/firefox-driver`
3. Install cypress `npm install -g cypress-cli`.
4. Install cypress app `cypress-cli install`
5. Start cypress server from cypress app in menubar

Going forward many of these steps will be automated away

1. the selenium server and driver will be wrapped by the test start
command
2. The cypress desktop app will be replaced by the cli tool

### CODE REVIEW

File: public/js/components/Frames.js [hunk 1]
```diff
@@ -27,18 +27,24 @@ function renderFrameLocation(frame) {
   return div({ className: "location" }, basename(frame.where.source.url));
 }
 
+function renderFrame(frame, selectedFrame) {
+
```
Comment (jlongster) [inline public/js/components/Frames.js]: nit: extra line here?

File: public/js/test/cypress/commands.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// ***********************************************
```
Comment (jlongster) [inline public/js/test/cypress/commands.js]: These files are all commented out, are they templates or going to be fleshed out later?

**Comment by jlongster** [comment]
r+ once circleci passes. It's not clear how to actually run this, can you add either docs or a package.json script? Also, eventually I think we should share the CSS selectors used instead of embedding them in the test.

File: public/js/test/cypress/commands.js (hunk 2 above)
Comment (jasonLaster) [inline public/js/test/cypress/commands.js]: templates. not sure if we'll use these

**Comment by jasonLaster** [comment]
@jlongster I'll write docs / steps soon

**Comment by jasonLaster** [comment]
merged

**Pull request closed without merging**